### PR TITLE
fix: App crashes in User Profile Section #729 [Updated ]

### LINKED ISSUE #729: App crashes in User Profile Section
**Summary:**

App crashes on opening User Profile Section

**Steps to reproduce:**

Open User Profile from Navigation Drawer by clicking on Personal Image

**Expected behaviour:**

App should not crash

**Observed behaviour:**

App crashes

**Device and Android version:**

Xiaomi

**Screenshots:**

issue_user_profile_crashes

### PR DESCRIPTION
**Pull request by ankurs287**
Fixes #729 #753 

Reason behind crash:

When there is no entry in database for client's group.
When there is no entry in database for client's dob.

### CODE REVIEW

**Comment by ankurs287** [comment]
@therajanmaurya Hey, I have updated the code as you told. Kindly, look into it. Thanks

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/UserProfileFragment.java [hunk 1]
```diff
@@ -153,16 +154,26 @@ public void showUserDetails(Client client) {
         tvGroups.setText(getGroups(client.getGroups()));
         tvClientClassification.setText(client.getClientClassification().getName());
         tvPhoneNumber.setText(client.getMobileNo());
-        tvDOB.setText(DateHelper.getDateAsString(client.getDobDate()));
+        if (client.getDobDate().size() != 3) {  // no data entry in database for the client
+            tvDOB.setText("No DOB found.");
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/ui/fragments/UserProfileFragment.java]: @ankurs287 we are not using any hardcoded string anywhere. So move this string to `strings.xml` and you string value must be `No Date of Birth`

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/UserProfileFragment.java [hunk 2]
```diff
@@ -153,16 +154,26 @@ public void showUserDetails(Client client) {
...
 
     /**
      * Generate a string for groups which the client is part of.
+     *
      * @param groups {@link List} of {@link Group} which client is a part of.
      * @return Returns String of groups
      */
     private String getGroups(List<Group> groups) {
+        if (groups.size() == 0) {
+            return "No groups found."; // no groups entry in database for the client
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/ui/fragments/UserProfileFragment.java]: @ankurs287 we are not using any hardcoded string anywhere. So move this string to `strings.xml` and you string value must be `Not Assigned with any group`.

File: app/src/main/java/org/mifos/mobilebanking/utils/DateHelper.java [hunk 3]
```diff
@@ -26,12 +27,10 @@
 
     /**
      * the result string uses the list given in a reverse order ([x, y, z] results in "z y x")
-     *
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/utils/DateHelper.java]: please revert these changes.

File: app/src/main/java/org/mifos/mobilebanking/utils/DateHelper.java [hunk 4]
```diff
@@ -26,12 +27,10 @@
...
      * @param integersOfDate [year-month-day] (ex [2016, 4, 14])
      * @return date in the format day month year (ex 14 Apr 2016)
      */
     public static String getDateAsString(List<Integer> integersOfDate) {
-
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/utils/DateHelper.java]: please revert these changes.

File: app/src/main/java/org/mifos/mobilebanking/utils/DateHelper.java [hunk 5]
```diff
@@ -40,7 +39,6 @@ public static String getDateAsString(List<Integer> integersOfDate) {
                 .append(integersOfDate.get(0));
 
         return stringBuilder.toString();
-
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/utils/DateHelper.java:43]: please revert these changes.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/UserProfileFragment.java (hunk 1 above)
Comment (ankurs287) [reply app/src/main/java/org/mifos/mobilebanking/ui/fragments/UserProfileFragment.java]: okay. I will do that.

File: app/src/main/java/org/mifos/mobilebanking/utils/DateHelper.java (hunk 5 above)
Comment (ankurs287) [reply app/src/main/java/org/mifos/mobilebanking/utils/DateHelper.java:43]: ok

**Review by therajanmaurya — APPROVED** [review]

**Pull request merged**